src/bp/ui-studio/src/web/views/Config/index.tsx

```diff
@@ -0,0 +1,465 @@
+import { Button, Callout, Card, Collapse, Elevation, FileInput, FormGroup, InputGroup, Intent } from '@blueprintjs/core'
+import axios from 'axios'
+import { any } from 'bluebird'
+import { BotConfig } from 'botpress/sdk'
+import { BotEditSchema } from 'common/validation'
+import Joi from 'joi'
+import _ from 'lodash'
+import React, { Component } from 'react'
+import Select from 'react-select'
+import confirmDialog from '~/components/Shared/ConfirmDialog'
+import { toastFailure, toastSuccess } from '~/components/Shared/Utils/Toaster'
+
+import style from './style.scss'
+
+const statusList: SelectItem[] = [
+  { label: 'Published', value: 'public' },
+  { label: 'Collaborators Only', value: 'private' },
+  { label: 'Unmounted', value: 'disabled' }
+]
+
+const axiosConfig = {
+  baseURL: 'api/v1/'
+}
+
+interface StateBot {
+  name: string
+  status: string
+  description: string
+  selectedDefaultLang: string
+  selectedLanguages: SelectItem[]
+  website: string
+  phoneNumber: string
+  emailAddress: string
+  termsConditions: string
+  privacyPolicy: string
+  avatarUrl: string
+  coverPictureUrl: string
+}
+
+interface StateVars {
+  botId: string
+  licensing: Licensing
+  languages: SelectItem[]
+  statuses: SelectItem[]
+  error: any
+  isSaving: boolean
+  isDetailsOpen: boolean
+  isPicturesOpen: boolean
+}
+
+type State = StateBot & StateVars
+
+interface Licensing {
+  isPro: boolean
+}
+
+interface SelectItem {
+  label: string
+  value: string
+}
+
+class ConfigView extends Component<any, State> {
+  initialFormState: StateBot = {
+    name: '',
+    status: '',
+    description: '',
+    selectedDefaultLang: '',
+    selectedLanguages: [],
+    website: '',
+    phoneNumber: '',
+    emailAddress: '',
+    termsConditions: '',
+    privacyPolicy: '',
+    avatarUrl: '',
+    coverPictureUrl: ''
+  }
+
+  state = {
+    botId: window.BOT_ID,
+    ...this.initialFormState,
+    licensing: undefined,
+    languages: [],
+    statuses: statusList,
+    error: undefined,
+    isSaving: false,
+    isDetailsOpen: false,
+    isPicturesOpen: false
+  }
+
+  async componentDidMount() {
+    const bots = await this.fetchBots()
+    const languages = await this.fetchLanguages()
+    const licensing = await this.fetchLicensing()
```

Comment:
Can you use the redux store for this? It will prevent having to call the backend multiple times when playing around on the studio (ex: going back and forth to the config page)